src/components/layouts/php-collection-page-layout.tsx

```diff
@@ -120,35 +135,36 @@ const ModuleCollection = ({module, children}: {module: any; children: any}) => {
   )
 }
 
-type NestedResource = {
-  type: 'module'
-  resourceList: ResourceList
-  title: string
-  description: string
-}
-type ResourceList = Array<NestedResource | LessonResource>
-
 const ResourceCollection = ({
   resourceList,
   completedLessonSlugs,
 }: {
-  resourceList: ResourceList
+  resourceList: Resource[]
   completedLessonSlugs: string[]
 }) => {
   return (
     <>
-      {resourceList.map((resource: any, index: number) => {
+      {resourceList.map((resource: Resource, index: number) => {
+        if ('resourceList' in resource) {
```

Comment:
I added a type guard here since some types that `resource` could be won't contain a `resourceList` since they're either an existing type or in the future an "atomic" type like `videoResource`

Reply:
suggestion: use a type that would show you an error on the calling side:

`(resourceWithList: {type: 'module' | 'lesson', resourceList: Resource[]}, index: number)`

otherwise you might as well switch back to `any` imo because this isn't providing any type benefit and adds conditional complexity